io/jvm/src/main/scala/fs2/io/net/SelectingSocket.scala

```diff
@@ -77,7 +79,61 @@ private final class SelectingSocket[F[_]: LiftIO] private (
     F.delay {
       ch.shutdownInput(); ()
     }
+  override def sendfile(
+      file: FileHandle[F],
+      offset: Long,
+      count: Long,
+      chunkSize: Int
+  ): Stream[F, Nothing] = file match {
+    case syncFileHandle: SyncFileHandle[F] =>
+      val fileChannel = syncFileHandle.chan
 
+      def go(currOffset: Long, remaining: Long): F[Unit] =
+        if (remaining <= 0) F.unit
+        else {
+          val toTransfer = remaining.min(chunkSize.toLong)
+          F.blocking(fileChannel.transferTo(currOffset, toTransfer, ch)).flatMap { written =>
```

Comment:
We don't need to use `chunkSize` here, we should leave it to the underlying syscall to chunk/buffer things appropriately.

```suggestion
          F.blocking(fileChannel.transferTo(currOffset, remaining, ch)).flatMap { written =>
```

Reply:
@armanbilge made the required change.